### PR TITLE
Promoting Matrix to Beta 🎉 🎓 

### PR DESCRIPTION
**Pull request by pritidesai**
# Changes

Matrix was introduced in [Tekton Pipelines 0.38](https://github.com/tektoncd/pipeline/releases/tag/v0.38.0). We have introduced a lot of testing around matrix feature including unit/e2e/integration tests based on the feedback from the users and community. Matrix unblocks variety of use cases migrating from Travis/Jenkins.

This is a very crucial feature for many end-users while allowing to maintain clean, concise, and flexible catalog of pipelines.

Part of https://github.com/tektoncd/pipeline/issues/5265 and https://github.com/tektoncd/pipeline/issues/6110

We have accomplished most of the checklist except [producing task results](https://github.com/tektoncd/pipeline/issues/5266). The implementation for this feature is nearly done and opening a [PR](https://github.com/tektoncd/pipeline/compare/main...EmmaMunley:tep-140-produce-results-matrix) soon.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Matrix feature is promoted to beta and possible to utilize with enable-api-fields set to beta.
```

### CODE REVIEW

**Comment by JeromeJu** [comment]
Thanks a lot @pritidesai . Excited to take a look. 
/assign
Would we want this to be added to the current milestone?

File: pkg/apis/pipeline/v1beta1/pipeline_validation.go [hunk 1]
```diff
@@ -182,9 +182,6 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 func (pt *PipelineTask) validateMatrix(ctx context.Context) (errs *apis.FieldError) {
 	if pt.IsMatrixed() {
-		// This is an alpha feature and will fail validation if it's used in a pipeline spec
-		// when the enable-api-fields feature gate is anything but "alpha".
-		errs = errs.Also(version.ValidateEnabledAPIFields(ctx, "matrix", config.AlphaAPIFields))
```
Comment (JeromeJu) [inline pkg/apis/pipeline/v1beta1/pipeline_validation.go:187]: Would we still want the beta validations for matrix in v1beta1?
My understanding for now (without TEP0138 being merged and the API-compatibility policy being updated), we might still want the feature versioning and API version to be decoupled. I wonder if this validation could also be added as the same for v1. <referring @afrittoli 's previous [beta promotion of isolated steps(TEP0029)](https://github.com/tektoncd/pipeline/pull/6922)>

Comment (pritidesai) [reply pkg/apis/pipeline/v1beta1/pipeline_validation.go:187]: As per the discussion in the API WG, just to be consistent, we are adding such validation where `v1beta1` specifications are validated against `enable-api-fields`. Thanks!

Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/pipeline_validation.go:187]: Thanks @pritidesai !

**Comment by jerop** [comment]
> 'The step "echo" in TaskRun "matrixed-pr-sp4hz-platforms-and-browsers-dag-0"
              failed to pull the image "". The pod errored with the message: "Back-off pulling
              image "alpine"."'

/test pull-tekton-pipeline-beta-integration-tests

**Review by vdemeester — APPROVED** [review]

File: docs/additional-configs.md [hunk 2]
```diff
@@ -332,13 +331,14 @@ except where otherwise noted.
...
+| [Isolated `Step` & `Sidecar` `Workspaces`](./workspaces.md#isolated-workspaces)                     | [TEP-0029](https://github.com/tektoncd/community/blob/main/teps/0029-step-workspaces.md)                                   | [v0.24.0](https://github.com/tektoncd/pipeline/releases/tag/v0.24.0) | [v0.50.0](https://github.com/tektoncd/pipeline/releases/tag/v0.50.0) |                                                                      | Yes                                             |
+| [Matrix](./matrix.md)                                                                               | [TEP-0090](https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md)                                            | [v0.38.0](https://github.com/tektoncd/pipeline/releases/tag/v0.38.0) |                               | [v0.53.0](https://github.com/tektoncd/pipeline/releases/tag/v0.53.0) | No                                              |
```
Comment (Yongxuanzhang) [inline docs/additional-configs.md:341]: The beta release version is under individual flag column

**Review by Yongxuanzhang — APPROVED** [review]

File: docs/additional-configs.md (hunk 2 above)
Comment (pritidesai) [reply docs/additional-configs.md]: thanks @Yongxuanzhang for catching this, fixed it!

**Pull request merged**